cypress_test/integration_tests/desktop/impress/undo_redo_spec.js

```diff
@@ -0,0 +1,66 @@
+/* global describe it cy beforeEach require afterEach*/
+
+var helper = require('../../common/helper');
+
+describe('Editing Operations', function() {
+	var testFileName = 'comment_switching.odp';
+
+	beforeEach(function() {
+		helper.beforeAll(testFileName, 'impress');
+	});
+
+	afterEach(function() {
+		helper.afterAll(testFileName, this.currentTest.state);
+	});
+
+
+	function undo() {
+		cy.get('.leaflet-layer').dblclick();
+
+		helper.typeIntoDocument('Hello World');
+
+		helper.typeIntoDocument('{ctrl}z');
+
+		helper.selectAllText();
+
+		helper.expectTextForClipboard('Hello Worl');
+	}
+
+	it('Undo', function() {
+		undo();
+	});
+
+
+	it('Redo', function() {
+		undo();
+		helper.typeIntoDocument('{ctrl}y');
+
+		helper.selectAllText();
+
+		helper.expectTextForClipboard('Hello World');
+	});
+
+	it('Repair Document', function() {
+		cy.get('.leaflet-layer').dblclick();
+
+		helper.typeIntoDocument('Hello World');
+
+		helper.selectAllText();
```

Comment:
this is not needed I think.

Reply:
what are you talking about ? If you are talking about dblclick, it  is needed to get cursor

Reply:
talking about `selectAllText`